### DISCUSSION TITLE
Enable the HueSaturation on ExtendedColorLightDevice

### DISCUSSION THREAD

**Discussion by MatthieuCa** [matter-js/matter.js#1013] [Q&A] [answered]
Hello everyone !

I am facing an issue, I am trying to enable the HueSaturation on the ExtendedColorLightDevice by taking exemple on the #996 discussion but it doesn't work at all for me. 

When I add the code:
`MatterBridgeLightEndpoint(ExtendedColorLightRequirements.ColorControlServer.with('HueSaturation'))`

I have this error:
`The 'this' context of type 'Type<WithAlterations<WithFeatures<Of<{ readonly id: 768; readonly name: "ColorControl"; readonly revision: 5; readonly features: { readonly hueSaturation: BitFlag; readonly enhancedHue: BitFlag; readonly colorLoop: BitFlag; readonly xy: BitFlag; readonly colorTemperature: BitFlag; }; readonly attributes: { ...; }; r...' is not assignable to method's 'this' of type 'Type<ClusterType, Type, ClusterInterface<{}>>'.
  Types of property 'cluster' are incompatible.
    Type 'WithAlterations<WithFeatures<Of<{ readonly id: 768; readonly name: "ColorControl"; readonly revision: 5; readonly features: { readonly hueSaturation: BitFlag; readonly enhancedHue: BitFlag; readonly colorLoop: BitFlag; readonly xy: BitFlag; readonly colorTemperature: BitFlag; }; readonly attributes: { ...; }; readon...' is not assignable to type 'ClusterType'`

Have you any idea how to resolve this ?

Thank you for your help !

**Comment by MatthieuCa** [comment]
I made a mistake on what I give you, here is the correct code:

```javascript
const endpoint = new Endpoint(
    MatterBridgeLightEndpoint.with(ExtendedColorLightRequirements.ColorControlServer.with('HueSaturation')),
    {
      id: 'LightDimmerColor',
      onOff: {
        onOff: false,
      },
      levelControl: {
        currentLevel: 254,
        options: {
          executeIfOff: true,
        },
      },
      colorControl: {
        startUpColorTemperatureMireds: 370,
        coupleColorTempToLevelMinMireds: 157,
      },
    },
  )
```

**Comment by MatthieuCa** [comment, marked as answer]
Found the answer, I have a to desactivate exactOptionalPropertyTypes on my tsconfig
```
"exactOptionalPropertyTypes": false
```

**Comment by Apollon77** [comment]
What is MatterBridgeLightEndpoint? Cna oyu please show the fiull info? Normally this tsconfig setting should not be needed.

When I do 

```javascript
const endpoint = new Endpoint(
    ExtendedColorLightDevice.with(ExtendedColorLightRequirements.ColorControlServer.with("HueSaturation")),
    {
        id: "LightDimmerColor",
        onOff: {
            onOff: false,
        },
        levelControl: {
            currentLevel: 254,
            options: {
                executeIfOff: true,
            },
        },
        colorControl: {
            // startUpColorTemperatureMireds: 370,
            // coupleColorTempToLevelMinMireds: 157,
        },
    },
);
```

Then TS only tells me that the two "Mireds" attrinutes you add in the defaults are not allowed because when you just enable HueSaturation feature then the ColorTemperature settings are not allowed

**Reply by Apollon77** [reply]
In fact if you want to add features you can do it that way, but then you need to provide all features because you basically overwrite the full cluster on the device, so all target features needs to be set.
There are also options to dynamically enable single features

**Reply by MatthieuCa** [reply]
I need to add this tsconfig due to my working environment. I have a lot or rules and this one was enable so this is why I have the error.
The MatterBridgeEndpoint is an override Device.

**Reply by Apollon77** [reply]
And what exactly you override there?

**Reply by MatthieuCa** [reply]
The ExtendedColorLightDevice

**Reply by Apollon77** [reply]
Hm ok .. but still the color temperature attributes are invalid when you not also enable Color Temperature feature

**Reply by MatthieuCa** [reply]
Yes this is what  I want I delete the parameter right after I post this
